auth/basic.go

```diff
@@ -0,0 +1,66 @@
+package auth
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/rancher/go-rancher/api"
+)
+
+const rancherID = "rancher_id:%v"
+
+func basicAuth(w http.ResponseWriter, r *http.Request) error {
+	username, password, authFound := r.BasicAuth()
+	if !authFound {
+		return fmt.Errorf("No basic auth found.")
+	}
+	accountID := getAccount(username, password)
```

Comment:
don't rely on account id 0 being bad, return a (int64, bool) where the bool is `ok`